libcomps/src/python/src/pycomps.c

```diff
@@ -345,6 +345,10 @@ PyObject* PyCOMPS_fromxml_str(PyObject *self, PyObject *args, PyObject *kwds) {
         return NULL;
     }
     parsed_ret = comps_parse_str(parsed, tmps, options);
+    if (parsed_ret == -1) {
+        PyErr_SetString(PyCOMPSExc_ParserError, "Fatal parser error");
+        return NULL;
+    }
```

Comment:
Are you sure that the movement will not create memory leak? There are several object de-alocated after the return.

Reply:
`options` value could be the first candidate.

Reply:
Oh right, in case of an error it was causing a memory leak.
I have updated it, please take a look.